nautobot/dcim/api/serializers.py

```diff
@@ -797,13 +797,19 @@ class Meta:
     def validate(self, data):
         # Validate many-to-many VLAN assignments
         device = self.instance.device if self.instance else data.get("device")
-        # TODO: after Location model replaced Site, which was not a hierarchical model, should we allow users to assign a VLAN belongs to
-        # the parent Location or the child location of `device.location`?
+        location = None
+        if device:
+            location = device.location
+        if location:
+            location_ids = [ancestor.id for ancestor in location.ancestors()]
+            location_ids.append(location.id)
```

Comment:
```suggestion
            location_ids = location.ancestors(include_self=True).values_list("id", flat=True)
```

Reply:
It occurs to me we should probably add an `.ancestor_ids()` method to TreeQuerySet as well, as this is a bit recursive - `.ancestors()` gets the list of PKs itself and then uses that list to derive a queryset, but here we then use the queryset to get the list of PKs. Not blocking for this PR though.